backend/src/main/java/com/capstone/moneytree/facade/EmailSender.java

```diff
@@ -0,0 +1,71 @@
+
+import com.capstone.moneytree.model.node.User;
+import net.jacobpeterson.domain.alpaca.order.Order;
+import net.jacobpeterson.domain.alpaca.streaming.trade.TradeUpdate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Component;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.time.ZonedDateTime;
+
+@Component
+public class EmailSender {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSender.class);
+
+    @Autowired
+    private JavaMailSender mailSender;
+
+    public void sendOrderCompletedEmail(User user, TradeUpdate trade, String subject) {
+        String username = user.getUsername();
+        String toEmail = user.getEmail();
+        String text = setOrderCompletedTemplate(username, trade);
+        send(toEmail, subject, text);
+    }
+
+    private void send(String to, String subject, String text) {
+        MimeMessage mimeMessage = mailSender.createMimeMessage();
+        try {
+            MimeMessageHelper messageHelper = new MimeMessageHelper(mimeMessage, true);
+            String from = System.getenv().get("SPRING_BOOT_EMAIL");
+            messageHelper.setFrom(from);
+            messageHelper.setTo(to);
+            messageHelper.setSubject(subject);
+            messageHelper.setText(text);
+
+            mailSender.send(mimeMessage);
+            LOGGER.info("Email sent to {}", to);
+        } catch (MessagingException e) {
+            LOGGER.error("Error sending email: {}", e.getMessage());
+        }
+    }
+
+    private String setOrderCompletedTemplate(String username, TradeUpdate trade) {
+        Order order = trade.getOrder();
+        String stockName = order.getSymbol();
+        String quantity = order.getQty();
+        String action = order.getType();
+        String avgPrice = order.getFilledAvgPrice();
+        ZonedDateTime timestamp = trade.getTimestamp();
+        String totalAmount = trade.getPrice();
+
+        return "Hello " + username + ",\n" +
+                "Thank you for trading with us. Your order details are indicated below.\n" +
+                "If you would like to view the status of your order, please visit http://money-tree.tech.\n\n" +
+                "Stock Purchased: " + stockName + "\n" +
+                "Quantity: " + quantity + "\n" +
+                "Average Share Price: " + avgPrice + "\n" +
+                "Time of Purchase: " + timestamp.getMonth().getValue() + "/" + timestamp.getDayOfMonth() + "/" + timestamp.getYear() +
+                " at " + timestamp.getHour() + ":" + timestamp.getMinute() + ":" + timestamp.getSecond() + "\n" +
+                "Total Amount: " + totalAmount + "\n" +
+                "Action: " + action + "\n\n" +
```

Comment:
This is just a suggestion: this could be in a class method (like toString()), but since we didn't write the Order class, I'm thinking maybe a modified version of the visitor design pattern (something like a helper class if you like). It would allow you to write orderVisitor.visit(order). In that visitor class, there could also be other methods for different printing structure like logVisit for logging or debugVisit for debugging.

Reply:
Yeah I wanted to have a separate Mail class that contains all the necessary information for it. I'll see what I can do.

Reply:
Thanks for the comment Walter! I have created a new Mail class object and refactored everything inside it. Also created a toString() method that is used for the construction of the email.

Reply:
That would be ideal for a refactoring activity!

Reply:
that's exactly what I thought when commenting this. Who would've thought refactoring would make us all excited :P.